Centos based image-stream for usage in OpenShift

sspeiche opened the pull request.

sspeiche commented:
@bparees fyi

sspeiche commented:
@hhorak I wasn't sure naming convention for these repos but I figured this could just be a start. I can update library and pull it into OSO starter as well. Just let me know.

jcajka commented:
Could you move the stream file in to separate directory, like imagestreams or a like? @hhorak Or is there any current convention for carrying the stream files in the SCL repositories.

Otherwise LGTM. I have verifies that the stream is usable while looking in to the issue #5.

sspeiche commented:
@jcajka I updated the structure to match https://github.com/sclorg/s2i-ruby-container/tree/master/imagestreams and switched the sample to use a dedicated separate repo for the example.

Will evolve the example as needed and move to sclorg (or someone can create better example)

pvalena commented:
Just a side note: having some standartized locations for template and imagestream files would be really nice (I do have some automation regarding that and intend to have more).

jcajka commented:
@sspeiche Thanks. If there will be agreement on some other location we can easily move it afterwards.
@pvalena What I have been looking through the SCL repos. They have the streamfiles folder. I guess you are expecting it right?

jcajka commented:
@pvalena has there been any progress on setting some standard for streamfiles (location)?
I would really like to merge this soon.

pvalena commented:
@jcajka the current location is fine.
Thanks!

pvalena commented on imagestreams/golang-centos7.json:
```diff
@@ -0,0 +1,47 @@
+                    "kind": "ImageStreamTag",
+                    "name": "1.8"
+                },
+                "name": "latest"
```
You're missing:
```

        "referencePolicy": {
          "type": "Local"
        }
```
here. (See [example](https://github.com/sclorg/s2i-ruby-container/blob/master/imagestreams/ruby-centos7.json#L27))

pvalena commented on imagestreams/golang-centos7.json:
```diff
@@ -0,0 +1,47 @@
+                    "kind": "DockerImage",
+                    "name": "docker.io/centos/go-toolset-7-centos7:latest"
+                },
+                "name": "1.8"
```
Same here.

pvalena commented:
I've not tested it (esp. the example), and there're no tests. Could you provide sample of logs for 3.9?

sspeiche commented:
@pvalena I think I've made all needed changes and is ready to merge, let me know if you need anything else

pvalena commented on imagestreams/golang-centos7.json:
```diff
@@ -0,0 +1,53 @@
+                    "iconClass": "icon-go-gopher",
+                    "openshift.io/display-name": "Go (Latest)",
+                    "openshift.io/provider-display-name": "Red Hat, Inc.",
+                    "sampleRepo": "https://github.com/sspeiche/golang-ex.git",
```
Would it be possible to move sample app into https://github.com/sclorg/ org?

bparees replied:
@pvalena i created https://github.com/sclorg/golang-ex
feel free to update the settings/collaborators on the repo to whatever you guys want them to be.

pvalena commented on imagestreams/golang-centos7.json:
```diff
@@ -0,0 +1,53 @@
+                  "iconClass": "icon-go-gopher",
+                  "openshift.io/display-name": "Go (1.8)",
+                  "openshift.io/provider-display-name": "Red Hat, Inc.",
+                  "sampleRepo": "https://github.com/openshift/origin.git",
```
I'm not sure about this one... Shouldn't it be the same url as above?

sspeiche commented:
Moved my sample to https//github.com/sclorg/golang-ex.git and updated references to it from this image-stream (in both locations).

/ptal @pvalena

pvalena commented:
> before we proceed with this, we should fix the sample so that it is built using the s2i builder, not embedding its own Dockerfile.

@bparees, I somehow missed this, would you mind to elaborate? Do you mean the sample app? The `sclorg` one uses s2i, right?

bparees commented:
> @bparees, I somehow missed this, would you mind to elaborate? Do you mean the sample app? The sclorg one uses s2i, right?

I don't think the sclorg one existed before now.

We just created it here:
https://github.com/sclorg/golang-ex

but yes, it now uses s2i.

sspeiche commented:
@pvalena looks like this is ready to merge, not outstanding comments

pvalena commented:
Although I've not tested this, it LGTM.
